public/app/core/components/RolePicker/RolePickerMenu.tsx

```diff
@@ -0,0 +1,585 @@
+  const styles = getSelectStyles(theme);
+  const customStyles = useStyles2(getStyles);
+
+  const onClearInternal = async () => {
+    if (onClear) {
+      onClear();
+    }
+  };
+
+  return (
+    <div className={customStyles.subMenu} aria-label="Role picker submenu">
+      <CustomScrollbar autoHide={false} autoHeightMax="300px" hideHorizontalTrack>
+        <div className={styles.optionBody}>
+          {options.map((option, i) => (
+            <RoleMenuOption
+              data={option}
+              key={i}
+              isSelected={
+                !!(
+                  option.uid &&
+                  (!!selectedOptions.find((opt) => opt.uid === option.uid) ||
+                    disabledOptions.find((opt) => opt.uid === option.uid))
+                )
+              }
+              disabled={!!(option.uid && disabledOptions.find((opt) => opt.uid === option.uid))}
+              onChange={onSelect}
+              hideDescription
+            />
+          ))}
+        </div>
+      </CustomScrollbar>
+      <div className={customStyles.subMenuButtonRow}>
+        <HorizontalGroup justify="flex-end">
+          <Button size="sm" fill="text" onClick={onClearInternal}>
+            Clear
+          </Button>
+        </HorizontalGroup>
+      </div>
+    </div>
+  );
+};
+
+interface RoleMenuOptionProps<T> {
+  data: Role;
+  onChange: (value: Role) => void;
+  isSelected?: boolean;
+  isFocused?: boolean;
+  disabled?: boolean;
+  hideDescription?: boolean;
+}
+
+export const RoleMenuOption = React.forwardRef<HTMLDivElement, React.PropsWithChildren<RoleMenuOptionProps<any>>>(
+  ({ data, isFocused, isSelected, disabled, onChange, hideDescription }, ref) => {
+    const theme = useTheme2();
+    const styles = getSelectStyles(theme);
+    const customStyles = useStyles2(getStyles);
+
+    const wrapperClassName = cx(
+      styles.option,
+      isFocused && styles.optionFocused,
+      disabled && customStyles.menuOptionDisabled
+    );
+
+    const onChangeInternal = (event: FormEvent<HTMLElement>) => {
+      if (disabled) {
+        return;
+      }
+      event.preventDefault();
+      event.stopPropagation();
+      onChange(data);
+    };
+
+    return (
+      <Tooltip content={data.description}>
+        <div ref={ref} className={wrapperClassName} aria-label="Role picker option" onClick={onChangeInternal}>
+          <Checkbox
+            value={isSelected}
+            className={customStyles.menuOptionCheckbox}
+            onChange={onChangeInternal}
+            disabled={disabled}
+          />
+          <div className={cx(styles.optionBody, customStyles.menuOptionBody)}>
+            <span>{data.displayName || data.name}</span>
+            {!hideDescription && data.description && <div className={styles.optionDescription}>{data.description}</div>}
+          </div>
+        </div>
+      </Tooltip>
+    );
+  }
+);
+
+RoleMenuOption.displayName = 'RoleMenuOption';
+
+interface RoleMenuGroupsOptionProps<T> {
```

Comment:
```suggestion
interface RoleMenuGroupsOptionProps {
```
The generic type isn't used anywhere.